autoload/ale/handlers/solhint.vim

```diff
@@ -94,5 +95,5 @@ function! ale#handlers#solhint#GetCommand(buffer) abort
 
     return ale#node#Executable(a:buffer, l:executable)
     \   . (!empty(l:options) ? ' ' . l:options : '')
-    \   . ' --formatter compact %s'
+    \   . ' --formatter unix %s'
```

Comment:
Again there may be users tha cannot upgrade their solhint version and need the "compact" formatter so this cannot be easily removed.  Rather modify this linter to use "ale#semver#RunWithVersionCheck()" and use either compact or unix formatter depending on the version of solhint tool.

A good example on how to do this is the `ansible_lint.vim` linter.